bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -72,185 +73,168 @@ integration.define({
+                        indexChannels++
+                        continue
+                    }
 
-                        indexUser++
-                    } while (searchText.indexOf('>') >= 0)
-                    channelMessages[indexMessages].datetime = new Date(channelMessages[indexMessages].ts * 1000);
-                    parsedChannelMessages.push(channelMessages[indexMessages])
+                    channelMessages[indexMessages].text_changed = text.replace('<@' + user_ids[indexUser] + '>', userData.user.real_name)
 
-                    indexMessages++;
-                } while (channelMessages[indexMessages] !== undefined)
+                } else {
+                    throw new Error(`Could not retrieve user data (${resp.status}: ${resp.statusText})`);
+                }
 
-            } else {
-                throw new Error(`Could not retrieve favorite channel messages (${response.status}: ${response.statusText})`);
-            }
+                indexUser++
+            } while (searchText.indexOf('>') >= 0)
+            channelMessages[indexMessages].datetime = new Date(channelMessages[indexMessages].ts * 1000);
+            parsedChannelMessages.push(channelMessages[indexMessages])
 
-            indexChannels++;
-        } while (favoriteChannels[indexChannels] !== undefined)
+            indexMessages++;
+        } while (channelMessages[indexMessages] !== undefined)
 
-        const respChannels = await client.fetch('/conversations.list?types=public_channel,private_channel')
+        indexChannels++;
+    } while (favoriteChannels[indexChannels] !== undefined)
 
-        if (respChannels.ok) {
-            let allChannelsResponse = await respChannels.json()
-            let allChannels = allChannelsResponse.channels
-            let indexChannels = 0;
+    const respChannels = await client.fetch('/conversations.list?types=public_channel,private_channel')
 
-            do {
-                let channel = allChannels[indexChannels]
-                let respChannelsMembers = await client.fetch(`/conversations.members?channel=${channel.id}`)
+    if (!respChannels.ok) {
+        throw new Error(`Could not retrieve favorite channel messages (${respChannels.status}: ${respChannels.statusText})`);
+    }
 
-                if (respChannelsMembers.ok) {
-                    let allChannelsMember = await respChannelsMembers.json()
-                    let channelMembers = allChannelsMember.members
-                    if (channelMembers === undefined) {
-                        indexChannels++
-                        continue
-                    }
-                    let indexChannelsMembers = 0;
+    let allChannelsResponse = await respChannels.json()
+    let allChannels = allChannelsResponse.channels
+    let indexAllChannels = 0;
 
-                    do {
+    do {
+        let channel = allChannels[indexAllChannels]
+        let respChannelsMembers = await client.fetch(`/conversations.members?channel=${channel.id}`)
```

Comment:
Is it ok that you don't handle the error state?

Reply:
Added error handling.